lib/Sema/CSSimplify.cpp

```diff
@@ -6970,7 +6971,8 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     auto solveWithNewBaseOrName = [&](Type baseType,
                                       DeclNameRef memberName) -> SolutionKind {
-      return simplifyMemberConstraint(kind, baseType, memberName, memberTy,
+      return simplifyMemberConstraint(kind, baseType->lookThroughAllOptionalTypes(),
```

Comment:
I'm not sure silently stripping optionality here is the right solution, as it could cause us to miss recording useful optional unwrapping fixes.

Reply:
Ah, you are right here it is wrong, but that is not where it meant to be ... I'll adjust it :)